powerscale/provider/smb_share_settings_resource.go

```diff
@@ -0,0 +1,540 @@
+		response.Diagnostics.AddError(
+			"Error copying fields of smb share settings resource", err.Error(),
+		)
+		return
+	}
+	state.Zone = sharePlan.Zone
+	state.ID = types.StringValue("smb_share_settings_" + sharePlan.Zone.ValueString())
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	tflog.Info(ctx, "create smb share settings completed")
+}
+
+// Read reads the resource state.
+func (r SmbShareSettingsResource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	tflog.Info(ctx, "reading smb share settings")
+	var shareState models.SmbShareSettingsResourceModel
+	diags := request.State.Get(ctx, &shareState)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	zone := shareState.Zone.ValueString()
+	shareResponse, err := helper.GetSmbShareSettings(ctx, r.client, zone)
+	if err != nil {
+		errStr := constants.ReadSMBShareSettingsErrorMsg + "with error: "
+		message := helper.GetErrorString(err, errStr)
+		response.Diagnostics.AddError("Error reading smb share settings ",
+			message)
+		return
+	}
+
+	err = helper.CopyFieldsToNonNestedModel(ctx, shareResponse.Settings, &shareState)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Error copying fields of smb share settings resource", err.Error(),
+		)
+		return
+	}
+
+	shareState.Zone = types.StringValue(zone)
+	shareState.ID = types.StringValue("smb_share_settings_" + zone)
+	diags = response.State.Set(ctx, &shareState)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	tflog.Info(ctx, "read smb share settings completed")
+}
+
+// Update updates the resource state.
+func (r SmbShareSettingsResource) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
+	tflog.Info(ctx, "updating smb share settings")
+	var sharePlan models.SmbShareSettingsResourceModel
+	diags := request.Plan.Get(ctx, &sharePlan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	zone := sharePlan.Zone.ValueString()
+	var shareToUpdate powerscale.V7SmbSettingsShareSettings
+	// Get param from tf input
+	err := helper.ReadFromState(ctx, sharePlan, &shareToUpdate)
+	if err != nil {
+		errStr := constants.ReadSMBShareSettingsErrorMsg + "with error: "
+		message := helper.GetErrorString(err, errStr)
+		response.Diagnostics.AddError(
+			"Error updating smb share settings",
+			fmt.Sprintf("Could not read smb share settings param with error: %s", message),
+		)
+		return
+	}
+
+	err = helper.UpdateSmbShareSettings(ctx, r.client, shareToUpdate, zone)
+	if err != nil {
+		errStr := constants.UpdateSMBShareSettingsErrorMsg + "with error: "
+		message := helper.GetErrorString(err, errStr)
+		response.Diagnostics.AddError("Error updating smb share settings ",
+			message)
+		return
+	}
+
+	//scope := sharePlan.Scope.ValueString()
```

Comment:
hi @sapana05 ,seems the resource has `scope `param, why comment it here?

Reply:
hi @sapana05 , I found the scope param is added, thanks. Please remove this commented line.